configs/components/facter.rb

```diff
@@ -226,6 +224,23 @@
     ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
   end
 
+  if platform.is_osx?
+    ldd = "otool -L"
+  else
+    ldd = "ldd"
+  end
+
+  unless platform.is_windows? || platform.is_cross_compiled_linux? || platform.architecture == 'sparc'
+    pkg.check do
+      [
+        # Ensure we're linking against our libstdc++ and libgcc_s or statically linking
+        "#{ldd} lib/libfacter.so",
+        "[ $$(#{ldd} lib/libfacter.so | grep -c libstdc++)  -lt 1 ] || #{ldd} lib/libfacter.so | grep libstdc++ | grep -v ' /lib'",
+        "[ $$(#{ldd} lib/libfacter.so | grep -c libgcc_s) -lt 1 ] || #{ldd} lib/libfacter.so | grep libgcc_s | grep -v ' /lib'"
```

Comment:
Testing at build time, it looks like the path to `libstdc++` on most platforms (where we dynamically link) is `/opt/puppetlabs/puppet/lib`. However, there are some outliers (EL4, Ubuntu 15.10), where the path at build time is either `/opt/pl-build-tools/lib` or `/opt/pl-build-tools/lib64`.